builder/vmware/common/driver_mock.go

```diff
@@ -98,6 +135,30 @@ func (d *DriverMock) CommHost(state multistep.StateBag) (string, error) {
 	return d.CommHostResult, d.CommHostErr
 }
 
+func (d *DriverMock) HostAddress(state multistep.StateBag) (string, error) {
+	d.HostAddressCalled = true
+	d.HostAddressState = state
+	return d.HostAddressResult, d.HostAddressErr
```

Comment:
Where is this HostAddressResult value coming from?  I'm not seeing it in this function or anywhere else except where it is declared..

Reply:
Copy-paste issue. I didn't really understand how the mock-driver interface works and so I just copied the names and things from another mock-driver. ;)

Anyways, I added a new commit to this branch that attempts to auto-detect a hardware address from the first loopback-capable interface using the `net` library. If there's no interfaces (or an error with enumerating) then it builds a dummy interface with a null hardware address.